src/dotnet/ToolPackage/LockFileMatcher.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal class LockFileMatcher
+    {
+        private static string[] _pathInLockFilePathInArray;
+        private static string[] _entryPointPathInArray;
+
+        internal static bool MatchesFile(string pathInLockFile, string entryPoint)
+        {
+            _pathInLockFilePathInArray = SplitPathByDirectorySeparator(pathInLockFile);
+            _entryPointPathInArray = SplitPathByDirectorySeparator(entryPoint);
+
+            return EntryPointHasFileAtLeast()
+                   && PathInLockFileHasExcatlyToolsTfmRidAsParentDirectories()
+                   && RestMatchsOtherThanPathInLockFileHasToolsTfmRidAsParentDirectories();
+        }
+
+        private static bool RestMatchsOtherThanPathInLockFileHasToolsTfmRidAsParentDirectories()
```

Comment:
Nit: I think these method names are a bit too descriptive (I know they're private, but still); they read like test names.  I'd hate to have to type these out without IntelliSense.

Reply:
nit: this will be a philosophical debate. But I think method name length should be low priority here. Extreme case:
 [objective c naming](https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingMethods.html). And most editor has auto complete on words (by simply scan the existing doc and find words). Most "starter-pack" of vim and emacs will include such function as well.

Reply:
Agreed that naming can be a complete bikeshedding exercise (not to mention that good naming is difficult).  I simply prefer concise names that convey enough meaning; they don't need to be fully self-describing as that's what either documentation or the code itself is for.

Reply:
To me even the long name doesn't really help me understand what the method does.  So more than changing the name I would add comments to explain it.

Reply:
Please see if the local function + strong type input + comments is good enough